### PR TITLE
feat(core): include non-project files on fileMap and filesToProcess

### PR DESCRIPTION
**Pull request by AgentEnder**
## Current Behavior
`filesToProcess` doesn't include non project files, so there is no indication inside of `createDependencies` if one has changed. This leads to requiring custom caching and hashing behavior on the plugin side to avoid extra work.

## Expected Behavior
`filesToProcess` includes non project files

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/nx/src/config/project-graph.ts [hunk 1]
```diff
@@ -8,25 +8,60 @@ import { NxJsonConfiguration } from './nx-json';
...
+   * that string. If the element is a tuple, the first element inside of it
+   * is the target project, with the second element being the type of dependency.
+   */
+  deps?: (string | [target: string, type: string])[];
+}
+
+/**
+ * Represents metadata about a file that **does not** belong to a project.
+ */
+export interface NonProjectFileData extends BaseFileData {
```
Comment (FrozenPandaz) [inline packages/nx/src/config/project-graph.ts:32]: Make this more similar to the projectFileMap deps.

Add a tuple of `[src, target, type]`

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/nx/src/native/workspace/workspace_files.rs [hunk 2]
```diff
@@ -20,6 +20,11 @@ pub struct NxWorkspaceFiles {
     pub external_nodes: HashMap<String, JsObject>,
 }
 
+pub struct FileMap {
+    pub project_file_map: HashMap<String, Vec<FileData>>,
+    pub non_project_files: Vec<FileData>
+}
+
```
Comment (FrozenPandaz) [inline packages/nx/src/native/workspace/workspace_files.rs:27]: rm

File: packages/nx/src/project-graph/project-graph-builder.ts [hunk 3]
```diff
@@ -273,23 +324,28 @@ export class ProjectGraphBuilder {
...
-      const fileData = getFileData(
+      let fileData = getProjectFileData(
         source,
         sourceFile,
-        this.graph.nodes,
-        this.fileMap
+        this.projectFileMap
       );
+      const isProjectFileData = !!fileData;
+      fileData ??= getNonProjectFileData(sourceFile, this.allWorkspaceFiles);
```
Comment (FrozenPandaz) [inline packages/nx/src/project-graph/project-graph-builder.ts:333]:
```suggestion
      fileData ??= getNonProjectFileData(sourceFile, this.nonProjectFiles);
```

**Review by FrozenPandaz — APPROVED** [review]

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
Double check all of the createDependencies functions

**Review by FrozenPandaz — APPROVED** [review]

**Review by ndcunningham — APPROVED** [review]

**Pull request merged**